src/background.js

```diff
@@ -59,19 +67,22 @@ const stateManager = {
     // In other words, if the user has made their own decision for DoH,
     // then we want to respect that and never run the heuristics again
 
-    if (disableHeuristics) {
-      await stateManager.rememberTRRMode();
-      return false;
-    } else if ( prevMode !== curMode ||  curMode === 5 ||  curMode === 3) {
-      // Add logic specific if user disables DoH in about:config:
-      if ( curMode === 0 ) {
-        await stateManager.setState("manuallyDisabled");
+    // On Mismatch - run never run again (make init check a function)
+
+    if (prevMode !== curMode) {
+      log("Mismatched, curMode: ", curMode);
+      if (curMode === 0) {
```

Comment:
Do we care what the value is? Surely 5 and 3 are also important.

Reply:
Especially as the else case basically takes care of this within `prefHasUserValueCheck` do we still need anything other than this?

Reply:
We have to check for `0` (and apply `stateManager.rememberDisableHeuristics()` bc it's the default right now for network.trr.mode. When the mismatch is detected, we run the `trrModePrefHasUserValueCheck` function, but it doesn't disable heuristics as it appears that there's be no modification to the pref.